sktime/clustering/partitioning/_time_series_k_partition.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+"""Time Series K-Partition."""
+
+__author__ = ["Christopher Holder", "Tony Bagnall"]
+__all__ = ["TimeSeriesKPartition"]
+
+from typing import List
+import numpy as np
+
+from sklearn.metrics.pairwise import (
+    pairwise_distances_argmin_min,
+)
+from sktime.clustering.base.base_types import (
+    Metric_Parameter,
+    Metric_Function_Dict,
+    Numpy_Array,
+)
+from sktime.clustering.base.base import (
+    BaseCluster,
+    ClusterMixin,
+    Init_Algo,
+    Init_Algo_Dict,
+)
+from sktime.distances.elastic_cython import (
+    ddtw_distance,
+    dtw_distance,
+    erp_distance,
+    lcss_distance,
+    msm_distance,
+    twe_distance,
+    wddtw_distance,
+    wdtw_distance,
+)
+from sktime.clustering.base.base import BaseClusterCenterInitializer
+from sktime.clustering.partitioning._center_initializers import (
+    ForgyCenterInitializer,
+)
+from sktime.clustering.utils._utils import compute_pairwise_distances
+from sktime.distances.elastic import euclidean_distance
+
+
+class TimeSeriesKPartition(BaseCluster, ClusterMixin):
+    """
+    TODO:
+    Algorithm specific:
+
+    Need to implement more init algorithms i.e. (K means ++)
+
+    Probably want to rerun the algorithm multiple times and take the
+    best result as the init can be so impactful on final result
+
+    Multivariate support. While I've coded this with the intention to
+    support it I doubt it will work with multivariate so need to code
+    and improve that
+    """
+
+    __metric_dict: Metric_Function_Dict = {
```

Comment:
same as above, here and below
```suggestion
    _metric_dict: Metric_Function_Dict = {
```